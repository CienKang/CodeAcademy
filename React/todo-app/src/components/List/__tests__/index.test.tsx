import React from 'react';
import { fireEvent, render} from '@testing-library/react';
import { List } from '../index';

describe('List', () => {
    it('should render a list of tasks when some tasks are already present', () => {
        const setTasks = jest.fn();
        const { getByText } = render(<List tasks={['test1', 'test2']} setTasks={setTasks} />);

        expect(getByText('test1')).toBeInTheDocument();
        expect(getByText('test2')).toBeInTheDocument();
    });

    it('should delete the task when the delete button is clicked for a certain task', () => {
        const setTasks = jest.fn();

        const { getByTestId } = render(<List tasks={['test1', 'test2']} setTasks={setTasks} />);
        const deleteButton = getByTestId('0');
        fireEvent.click(deleteButton);

        expect(setTasks).toHaveBeenCalledWith(['test2']);
        expect(setTasks).not.toHaveBeenCalledWith(['test1']);
    });

    it('should match the snapshot when rendered for the first time with same tasks', () => {
        const setTasks = jest.fn();
        const { asFragment } = render(<List tasks={['test1', 'test2']} setTasks={setTasks} />);
        expect(asFragment()).toMatchSnapshot();
    });

});