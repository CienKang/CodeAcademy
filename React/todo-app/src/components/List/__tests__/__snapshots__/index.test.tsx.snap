// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should match the snapshot when rendered for the first time with same tasks 1`] = `
<DocumentFragment>
  <div
    class="todo-container"
  >
    <div>
      <h1>
        test1
      </h1>
      <button
        data-testid="0"
      >
         Delete 
      </button>
    </div>
    <div>
      <h1>
        test2
      </h1>
      <button
        data-testid="1"
      >
         Delete 
      </button>
    </div>
  </div>
</DocumentFragment>
`;
